plugin/pkg/scheduler/algorithm/predicates/predicates.go

```diff
@@ -1143,10 +1155,10 @@ func (c *PodAffinityChecker) getMatchingAntiAffinityTerms(pod *v1.Pod, allPods [
 
 // Checks if scheduling the pod onto this node would break any anti-affinity
 // rules indicated by the existing pods.
-func (c *PodAffinityChecker) satisfiesExistingPodsAntiAffinity(pod *v1.Pod, meta algorithm.PredicateMetadata, nodeInfo *schedulercache.NodeInfo) bool {
+func (c *PodAffinityChecker) satisfiesExistingPodsAntiAffinity(pod *v1.Pod, meta algorithm.PredicateMetadata, nodeInfo *schedulercache.NodeInfo) (bool, error) {
```

Comment:
Actually, thinking more about this, I guess this function and also `satisfiesPodsAffinityAntiAffinity` can return (algorithm.PredicateFailureReason, error). Then you return `ErrExistingPodsAntiAffinityRulesNotMatch` instead of false and an error or nil.